Framework/PythonInterface/plugins/algorithms/IntegratePeaksSkew.py

```diff
@@ -0,0 +1,714 @@
+    ebg_sq[ebg_sq == 0] = ebg_sq_avg
+    # subtract bg from focused peak
+    ypk = ypk - ybg  # removes background and powder lines (roughly line up in tof)
+    epk_sq = epk_sq + ebg_sq
+    return ypk, epk_sq
+
+
+def find_peak_mask(signal, ixlo, ixhi, irow, icol, use_nearest):
+    _, ipeak2D = find_bg_pts_seed_skew(signal[:, :, ixlo:ixhi].sum(axis=2).flatten())
+    non_bg_mask = np.zeros(signal.shape[0:2], dtype=bool)
+    non_bg_mask[np.unravel_index(ipeak2D, signal.shape[0:2])] = True
+    labeled_array, num_features = label(non_bg_mask)
+    # find label corresponding to peak
+    peak_label = labeled_array[irow, icol]
+    if peak_label == 0 and num_features > 0 and use_nearest:
+        irows, icols = get_nearest_non_masked_index(non_bg_mask, irow, icol)
+        # look for label corresponding to max. num peak pixels
+        labels = np.unique(labeled_array[irows, icols])
+        ilabel = np.argmax([np.sum(labeled_array == lab) for lab in labels])
+        peak_label = labels[ilabel]
+        # throw warning if distance is greater than half window?
+    peak_mask = labeled_array == peak_label
+    return peak_mask, non_bg_mask, peak_label
+
+
+class IntegratePeaksSkew(DataProcessorAlgorithm):
+
+    def name(self):
+        return "IntegratePeaksSkew"
+
+    def category(self):
+        return "Diffraction\\Reduction"
+
+    def seeAlso(self):
+        return ["IntegratePeaksMD", "IntegrateEllipsoids"]
+
+    def summary(self):
+        return "Integrates single-crystal peaks in a MatrixWorkspace by identifying the peak pixels in a window on " \
+               "the detector by minimising the skew of the points in the background. The TOF extent of the peak is " \
+               "determined by maximising I/sig for the peak pixels identified using the skew method. "
+
+    def PyInit(self):
+        # Input
+        self.declareProperty(MatrixWorkspaceProperty(name="InputWorkspace", defaultValue="", direction=Direction.Input,
+                                                     validator=WorkspaceUnitValidator("TOF")),
+                             doc="A MatrixWorkspace to integrate (x-axis must be TOF).")
+        self.declareProperty(IPeaksWorkspaceProperty(name="PeaksWorkspace",
+                                                     defaultValue="",
+                                                     direction=Direction.Input),
+                             doc="A PeaksWorkspace containing the peaks to integrate.")
+
+        #   window parameters
+        self.declareProperty(name="NRows", defaultValue=8, direction=Direction.Input,
+                             validator=IntBoundedValidator(lower=1),
+                             doc="Half number of rows in the window on detector.")
+        self.declareProperty(name="NCols", defaultValue=8, direction=Direction.Input,
+                             validator=IntBoundedValidator(lower=1),
+                             doc="Half number of columns in the window on detector.")
```

Comment:
I have renamed these parameters to e.g. `DRows` - to distinguish them from other parameters like `NRowMax` and have clarified the doc strings and documentation.

Reply:
The description is much more clear now thanks. What does the new "D" prefix on the variable names indicate?

Reply:
In my head `D` stands for delta - so I tend to use it when I have a value +/- delta - which I admit is a bit arbitrary )it could also refer to the total width i suppose) and not obvious to users...I guess I didn't want to have `N` prefix as other parameters with `N` refer to the total number of rows/columns in e.g. the mask. I'm open to any variable name suggestions - I'm not completely happy with `DRows` etc!

Reply:
OK I see. I would normally associate a delta with a floating point quantity so perhaps that doesn't feel quite right here. How about just adding HW for half width in somewhere eg NRowsHW and NColsHW?